senders/client_factory.go

```diff
@@ -0,0 +1,80 @@
+package senders
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+)
+
+// Option Wavefront client configuration options
+type Option func(*configuration)
+
+// Configuration for the direct ingestion sender
+type configuration struct {
+	Server string // Wavefront URL of the form https://<INSTANCE>.wavefront.com
+	Token  string // Wavefront API token with direct data ingestion permission
+
+	// Optional configuration properties. Default values should suffice for most use cases.
+	// override the defaults only if you wish to set higher values.
+
+	// max batch of data sent per flush interval. defaults to 10,000. recommended not to exceed 40,000.
+	BatchSize int
+
+	// size of internal buffers beyond which received data is dropped.
+	// helps with handling brief increases in data and buffering on errors.
+	// separate buffers are maintained per data type (metrics, spans and distributions)
+	// buffers are not pre-allocated to max size and vary based on actual usage.
+	// defaults to 50,000. higher values could use more memory.
+	MaxBufferSize int
+
+	// interval (in seconds) at which to flush data to Wavefront. defaults to 1 Second.
+	// together with batch size controls the max theoretical throughput of the sender.
+	FlushIntervalSeconds int
+}
+
+// NewSender creates Wavefront client
+func NewSender(wfURL string, setters ...Option) (Sender, error) {
+	cfg := &configuration{}
+
+	u, err := url.Parse(wfURL)
+	if err != nil {
+		return nil, err
+	}
+
+	if !strings.HasPrefix(strings.ToLower(u.Scheme), "http") {
+		return nil, fmt.Errorf("invalid schema '%s', only 'http' is supported", u.Scheme)
+	}
+
+	if len(u.User.String()) > 0 {
+		cfg.Token = u.User.String()
```

Comment:
To maintain consistency with other SDKs (Java, Python etc), let's not encode the token in the URL.

Java provides a couple of [builder methods](https://github.com/wavefrontHQ/wavefront-sdk-java/blob/0f88426e1f827989b79f85d739d9c08c8e7eb58b/src/main/java/com/wavefront/sdk/common/clients/WavefrontClient.java#L133). We can probably just have the constructor take a `server` and `token` strings as input. And add clear comments describing what the server needs to be for proxy vs direct.

Reply:
Java also support it on the URL 
`// Create a factory and add a client with the following URL format: "http://TOKEN@DOMAIN.wavefront.com"`

I can add the token as an `Option`, OK ?